libshaderc_spvc/src/spvcir_pass.cc

```diff
@@ -19,6 +19,43 @@
 namespace spvtools {
 namespace opt {
 
+void SpvcIrPass::make_constant_null(uint32_t id, uint32_t type) {
+  auto &constant_type = get<spirv_cross::SPIRType>(type);
+
+  if (constant_type.pointer) {
+    auto &constant = set<spirv_cross::SPIRConstant>(id, type);
+    constant.make_null(constant_type);
+  } else if (!constant_type.array.empty()) {
+    assert(constant_type.parent_type &&
+           "constant type parent shouldn't be empty");
+    uint32_t parent_id = ir_->increase_bound_by(1);
+    make_constant_null(parent_id, constant_type.parent_type);
+
+    assert(constant_type.array_size_literal.back() &&
```

Comment:
I think this code only handles a one-dimensional array.   You're only checking the last dimension here.
From the SPIRV-Cross representation, it looks like constant_type.array_size_literal might be a long vector, with as many elements as number of nesting levels of the array.  That might be a fair assumption, but it should be stated in the method comment, and checked in the code (not just an assert).